### PR TITLE
Etcd Load Generator

### PR DESCRIPTION
**Pull request by hroyrh**
Generates load for etcd standalone instance.

### CODE REVIEW

**Comment by hroyrh** [comment]
Generates load for etcd standalone instance.

**Comment by hroyrh** [comment]
Hi, I have written this etcd load generator in golang.
@xiang90 can you please take a look

**Comment by hroyrh** [comment]
@xiang90 , I have written this etcd load generator. Can you please take a look

**Comment by beni05** [comment]
egy új felhasználói megpróbál a nyitjára, ?

**Comment by jeremyeder** [comment]
Ping?

File: tools/v2benchmark/README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# ETCD Load Testing
```
Comment (philips) [inline tools/v2benchmark/README.md]: etcd please

File: tools/v2benchmark/README.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+Note : This test, right now, supports only a single node etcd cluster.
```
Comment (philips) [inline tools/v2benchmark/README.md]: Why not support multi machine?

**Comment by philips** [comment]
Thank you for sending a patch for this tool. I think it is useful to have benchmarking for etcd but there would need to be substantial changes to merge this.

1. The code doesn't follow Go coding practices and the entire patch needs to be ran through `go fmt`. I will point out a couple of places the code can be improved but I don't have time do a full review. I would suggest that you spend some time reading the etcd code base or the Go standard library to learn a bit more about Go coding style as a next step.

2. etcd performance characteristics are quite different on multi-machine, why can't this be multi-machine?

3. pmap is a Linux only utility and we have people using etcd on different platforms, perhaps we should make this a separate project for etcd-linux-bench?

I hope that all makes sense and we are interested in perf tools for etcd overall; we just want to make sure code we merge is stuff we can maintain and have others hack on over time.

File: tools/v2benchmark/etcd_load.go [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+
+
+func main() {
+    flag.Parse()
+
+    if *fhelp {
+        flag.Usage()
+        return
+    }
+    if *fcfg_file != "null"{
```
Comment (philips) [inline tools/v2benchmark/etcd_load.go]: You set the default to "null" when you could have just set it to nil. Then you can simply do:

```
if fcfg_file == nil {
    flag.Usage()
    fmt.Println("Please input the cfg file")
}
```

File: tools/v2benchmark/etcd_load.go [hunk 4]
```diff
@@ -0,0 +1,540 @@
...
+        }
+        results <- &result{
+            errStr:   errStr,
+            duration: elapsed,
+        }
+    } 
+    defer wg.Done()
+}
+
+// This function handles the create requests.
```
Comment (philips) [inline tools/v2benchmark/etcd_load.go]: Please use godoc style comments: http://blog.golang.org/godoc-documenting-go-code

**Comment by gyuho** [comment]
@hroyrh Please check out https://github.com/coreos/etcd/blob/master/hack/benchmark/bench.sh.
We already have v2 benchmarks with load generator.

**Comment by xiang90** [comment]
I am closing this due to low activity.

**Pull request closed without merging**